thrift/thrift0.13/src/main/java/com/linecorp/armeria/server/thrift/ThriftCallService.java

```diff
@@ -140,14 +141,24 @@ public RpcResponse serve(ServiceRequestContext ctx, RpcRequest call) throws Exce
                 TApplicationException.UNKNOWN_METHOD, "unknown method: " + call.method()));
     }
 
-    private static void invoke(
+    private void invoke(
             ServiceRequestContext ctx,
             Object impl, ThriftFunction func, List<Object> args, CompletableRpcResponse reply) {
 
         try {
             final TBase<?, ?> tArgs = func.newArgs(args);
             if (func.isAsync()) {
-                invokeAsynchronously(impl, func, tArgs, reply);
+                if (useBlockingTaskExecutor) {
+                    ctx.blockingTaskExecutor().execute(() -> {
```

Comment:
Note: I understood that unlike `gRPC` interceptors, `thrift` rpcDecorators won't be invoked from the `blockingTaskExecutor`.

Reply:
That is correct.
I left [this comment](https://github.com/line/armeria/pull/5619#issuecomment-2072106170) to call the Thrift RPC decorators from the `blockingTaskExecutor` but @trustin suggested using the event loop for executing RPC decorators.